### PR TITLE
Codenvy-27 Adopt token service to workspace sharing

### PR DESCRIPTION
**Pull request by mshaposhnik**

### CODE REVIEW

**Comment by mshaposhnik** [comment]
@skabashnyuk @akorneta

**Comment by skabashnyuk** [comment]
ok

File: core/codenvy-hosted-machine-authentication/src/main/java/com/codenvy/machine/authentication/server/MachineTokenPermissionsFilter.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        String action;
+        String workspaceId;
+
+        switch (methodName) {
+            case "getMachineToken": {
+                workspaceId = ((String)arguments[0]);
+                action = USE;
+                break;
+            }
+            case "getUser": {
```
Comment (benoitf) [inline core/codenvy-hosted-machine-authentication/src/main/java/com/codenvy/machine/authentication/server/MachineTokenPermissionsFilter.java]: this case, getUser() is not tested

File: core/codenvy-hosted-machine-authentication/src/main/java/com/codenvy/machine/authentication/server/MachineTokenPermissionsFilter.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+import org.eclipse.che.api.core.ApiException;
+import org.eclipse.che.api.core.ForbiddenException;
+import org.eclipse.che.commons.env.EnvironmentContext;
+import org.eclipse.che.commons.subject.Subject;
+import org.eclipse.che.everrest.CheMethodInvokerFilter;
+import org.everrest.core.Filter;
+import org.everrest.core.resource.GenericMethodResource;
+
+import javax.inject.Inject;
```
Comment (benoitf) [inline core/codenvy-hosted-machine-authentication/src/main/java/com/codenvy/machine/authentication/server/MachineTokenPermissionsFilter.java]: unused import

File: core/codenvy-hosted-machine-authentication/src/main/java/com/codenvy/machine/authentication/server/MachineTokenPermissionsFilter.java (hunk 1 above)
Comment (mshaposhnik) [inline core/codenvy-hosted-machine-authentication/src/main/java/com/codenvy/machine/authentication/server/MachineTokenPermissionsFilter.java]: Added test

**Comment by akorneta** [comment]
ok

**Pull request merged**